lib/Conversion/TorchToLinalg/Pooling.cpp

```diff
@@ -1616,6 +1618,631 @@ class ConvertAtenAdaptivePoolOp : public OpConversionPattern<OpTy> {
+    Value ub1 = rewriter.create<tensor::DimOp>(loc, input, 1);
+    Value step = rewriter.create<arith::ConstantIndexOp>(loc, 1);
+    SmallVector<Value> finalOutputShape = {ub0, ub1, pooledH, pooledW};
+    Value finalOutputTensor = rewriter.create<tensor::EmptyOp>(
+        loc, getAsOpFoldResult(finalOutputShape), resultElementType);
+    auto forLoop = rewriter.create<scf::ForOp>(
+        loc, lb, ub0, step, ValueRange{},
+        [&](OpBuilder &b1, Location loc, Value iv0, ValueRange args) {
+          auto forLoop = b1.create<scf::ForOp>(
+              loc, lb, ub1, step, ValueRange{},
+              [&](OpBuilder &b, Location loc, Value iv1, ValueRange args) {
+                // Step 1: Extract bounds for region of interest (roi)
+                OpFoldResult zeroAttr = b.getI64IntegerAttr(0);
+                OpFoldResult oneAttr = b.getI64IntegerAttr(1);
+                OpFoldResult twoAttr = b.getI64IntegerAttr(2);
+                OpFoldResult threeAttr = b.getI64IntegerAttr(3);
+                OpFoldResult fourAttr = b.getI64IntegerAttr(4);
+                OpFoldResult fiveAttr = b.getI64IntegerAttr(5);
+                // SmallVector<Value> offsetVals{iv0, zeroAttr};
+                // SmallVector<OpFoldResult> sizeVals{oneAttr, fiveAttr};
+                SmallVector<OpFoldResult> strideVals{oneAttr, oneAttr, oneAttr,
+                                                     oneAttr};
+                // Value extractRoiBounds = b.create<tensor::ExtractSliceOp>(
+                //     loc, rois, offsetVals, sizeVals, strideVals);
+                Value lowY = b.create<tensor::ExtractOp>(
+                    loc, rois, ValueRange{iv0, oneAttr});
+                Value lowX = b.create<tensor::ExtractOp>(
+                    loc, rois, ValueRange{iv0, twoAttr});
+                Value highY = b.create<tensor::ExtractOp>(
+                    loc, rois, ValueRange{iv0, threeAttr});
+                Value highX = b.create<tensor::ExtractOp>(
+                    loc, rois, ValueRange{iv0, fourAttr});
+
+                lowY = b.create<arith::MulFOp>(loc, lowY, spatialScaleVal);
+                lowX = b.create<arith::MulFOp>(loc, lowX, spatialScaleVal);
+                highY = b.create<arith::MulFOp>(loc, highY, spatialScaleVal);
+                highX = b.create<arith::MulFOp>(loc, highX, spatialScaleVal);
+
+                lowY = b.create<arith::SubFOp>(loc, lowY, offset);
+                lowX = b.create<arith::SubFOp>(loc, lowX, offset);
+                highY = b.create<arith::SubFOp>(loc, highY, offset);
+                highX = b.create<arith::SubFOp>(loc, highX, offset);
+
+                // Step 2: Extract region of interest using bounds
+                Value lowY_int = b.create<math::FloorOp>(loc, lowY);
+                Value lowX_int = b.create<math::FloorOp>(loc, lowX);
+                Value highY_int = b.create<math::CeilOp>(loc, highY);
+                Value highX_int = b.create<math::CeilOp>(loc, highX);
```

Comment:
llvm doesn't use underscores in variable names, see https://llvm.org/docs/CodingStandards.html#name-types-functions-variables-and-enumerators-properly and https://mlir.llvm.org/getting_started/DeveloperGuide/#style-guide